File: 74. 搜索二维矩阵/74.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        h = len(matrix)
        w = len(matrix[0])
        i = 0
        j = 0
        while(i<h and j<w):
            # print(matrix[i][j])
            if(matrix[i][j]==target):
                return True
            elif(i+1<h and matrix[i+1][j]<=target):
                i+=1
            elif(j+1<w and matrix[i][j+1]<=target):
                j+=1
            else:
                return False
        return False

File: 74. 搜索二维矩阵/74.rs
impl Solution {
    pub fn search_matrix(matrix: Vec<Vec<i32>>, target: i32) -> bool {
        let mut min = 0;
        let mut max = matrix.len()-1;
        let mut i:i32;
        let mut j:i32;
        while min!=max{
            let mid = (min+max+1)/2;
            if(target>=matrix[mid][0]){
                min = mid;
            }
            else{
                max = mid-1;
            }
        }
        let i = min;
        min = 0;
        max = matrix[0].len()-1;
        while(min!=max){
            let mid = (min+max+1)/2;
            if(target>=matrix[i][mid]){
                min = mid;
            }
            else{
                max = mid-1;
            }
        }
        let j = min;
        if(target==matrix[i][j]){
            return true;
        }
        else{
            return false;
        }
    }
}